### PR TITLE
Mgmt command to remove inactive person identifiers

### PR DESCRIPTION
**Pull request by VirginiaDooley**
Ref https://app.asana.com/0/1204880927741389/1206795142514583/f

### CODE REVIEW

File: ynr/apps/people/management/commands/remove_inactive_person_links.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                    if resp in [404, 500, 502, 503, 504]:
+                        self.stdout.write(
+                            f"Status code: {resp} for {person.name} {identifier.value}"
+                        )
+                        inactive_links.append(identifier)
+                        for i, ident in enumerate(person_identifiers):
+                            if ident == identifier:
+                                person_identifiers.pop(i)
+                                print(
+                                    "Inactive links found for {person.name} and deleted"
```
Comment (symroe) [inline ynr/apps/people/management/commands/remove_inactive_person_links.py:45]: This doesn't actually delete? Two options:

1. Just make a report of links to delete and ask humans to do it. This is slow, but allows for double checking. In this case, can you output as TSV so we can copy into a Google Sheet?
2. Delete them in code, and in that case you'll need to use CandidateBot to make the version history correct.

File: ynr/apps/people/management/commands/remove_inactive_person_links.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        person identifier urls return a 200 status code.
+        """
+        inactive_links = []
+        people = Person.objects.all()
+        for person in people:
+            person_identifiers = person.get_all_identifiers
+            person_identifiers = [
+                identifier
+                for identifier in person_identifiers
+                if identifier.value.startswith("http")
```
Comment (symroe) [inline ynr/apps/people/management/commands/remove_inactive_person_links.py:23]: We can't detect working Facebook links as sometime they give a 404 even when not logged in. I think we should exclude any Facebook URL (including I think they have `fb.com` or something?)

File: ynr/apps/people/management/commands/remove_inactive_person_links.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                        f.write("person\tidentifier\tvalue\tstatus_code\n")
+                        for link in inactive_links:
+                            f.write(
+                                f"{person.name}\t{identifier}\t{identifier.value}\t{resp}\n"
+                            )
+                            f.close()
+
+
+def is_facebook_url(self, url):
+    return "facebook" in url or "fb" in url
```
Comment (symroe) [inline ynr/apps/people/management/commands/remove_inactive_person_links.py:61]: Could you add the domain part here, so we don't get false positives? e.g `facebook.com`, `fb.com`

Comment (VirginiaDooley) [reply ynr/apps/people/management/commands/remove_inactive_person_links.py]: Done!

Comment (symroe) [reply ynr/apps/people/management/commands/remove_inactive_person_links.py]: Are you sure?

Comment (VirginiaDooley) [reply ynr/apps/people/management/commands/remove_inactive_person_links.py]: Now I am :)

File: ynr/apps/people/management/commands/identify_inactive_person_links.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                        for link in inactive_links:
+                            f.write(
+                                f"{person.name}\t{identifier}\t{identifier.value}\t{resp}\n"
+                            )
+                            f.close()
+
+
+def is_facebook_url(self, url):
+    sanitized_url = url.lower().replace(" ", "")
+    return "facebook.com" in sanitized_url or "fb.com" in sanitized_url
```
[the remark this answers is not in the document]
Comment (VirginiaDooley) [reply ynr/apps/people/management/commands/identify_inactive_person_links.py:62]: Thoughts on these? I'm not sure how else to do this.

[the remark this answers is not in the document]
Comment (symroe) [reply ynr/apps/people/management/commands/identify_inactive_person_links.py:62]: Yeah, I guess this is a reasonable automated catch.

Maybe (copy and pasted form the internet):

```py
from urllib.parse import urlparse

def get_domain(url):
    parsed_url = urlparse(url)
    domain = parsed_url.netloc
    return domain

# Example usage
url = "https://www.example.com/some/path?query=param"
domain = get_domain(url)
```

File: ynr/apps/people/management/commands/identify_inactive_person_links.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+
+def get_domain(url):
+    parsed_url = urlparse(url)
+    return parsed_url.netloc
+
+
+def is_facebook_url(url):
+    domain = get_domain(url)
+    return "facebook.com" in domain or "fb.com" in domain
```
[the remark this answers is not in the document]
Comment (VirginiaDooley) [reply ynr/apps/people/management/commands/identify_inactive_person_links.py:15]: I think in this case, the urls are already within our db so going through additional lengths to check them seems like a waste. If this was for a form accepting urls, I could see the need.

**Pull request merged**